File: 4_median_of_2_sorted_arrays.cpp
//https://leetcode.com/problems/median-of-two-sorted-arrays/description/
class Solution {
public:
    double findMedianSortedArrays(vector<int>& nums1, vector<int>& nums2) {
        std::vector<int> fin;
        // 1,3
        // 2
        int i = 0;
        int j = 0;
        while( i < nums1.size() && j < nums2.size() )
        {
                if (nums1[i] < nums2[j])
                {
                    fin.push_back(nums1[i]);
                    i++;                        

                }
                else
                {
                    fin.push_back(nums2[j]);
                    j++;    
                }
        }
        while ( i < nums1.size())
        {
            fin.push_back(nums1[i]);
            i++;
        }
        
        while ( j < nums2.size())
        {
            fin.push_back(nums2[j]);
            j++;
        }
        
                for (int i =0; i < fin.size(); i++)
        {
            cout << fin[i] << " ";
        }
        
        if (fin.size() % 2 == 0)
        {
            //even
            int size = ceil( (fin.size() - 1) /2);
            cout << "size is:" << size;
            int first = fin[size];
            int second = fin[size + 1];
            return ((double)first + (double)second)/2;
            
        }
        else
        {
            return fin[ (fin.size() - 1)/2 ];
        }
    }
};

File: 4_median_of_2_sorted_arrays.java
//https://leetcode.com/problems/median-of-two-sorted-arrays/
class Solution {
    public double findMedianSortedArrays(int[] nums1, int[] nums2) {
        int mergedLength = nums1.length + nums2.length;
        int [] merged = new int[mergedLength];
        int i = 0, j = 0, z = 0;
        
        
        while( i < nums1.length && j < nums2.length ) {
            if (z > mergedLength/2) {
                break;
            }
            
            if (nums1[i] <= nums2[j]) {
                merged[z++] = nums1[i++];
            }
            else {
                merged[z++] = nums2[j++];
            }
        }
        
        while ( i < nums1.length) {
            if ( z > mergedLength/2) {
                break;
            }
            merged[z++] = nums1[i++];
        }
        
        while (j < nums2.length) {
            if (z > mergedLength/2) {
                break;
            }
            
            merged[z++] = nums2[j++];
        }
        
        if (mergedLength % 2 == 0 ) {
            return ((merged[z-1] + merged[z-2])/2.0);
        }
        else {
            return merged[z-1];
        }
    }
}